src/utils.js

```diff
@@ -127,6 +127,54 @@ function pick(o, props = []) {
     }, {});
 }
 
+/**
+ * Utility function used to perform a dfs search over a graph, recursively.
+ * @param {Array.<Object>} links - list of links connecting the nodes in a graph.
+ * @param {number} nodeid - id property of the node currently checking neighbors of.
+ * @param {Object} visited - Object keeping track of all the nodes previously visited.
+ * @param {Object} recStack - Object keeping track of recursion stack, containing the back edges.
+ * @returns {boolean} boolean reflecting wether the input nodes and links, results in a cyclic graph.
+ * @memberof utils
+ */
+const checkCyclicUtil = (links, nodeid, visited, recStack) => {
+    if (!visited[nodeid]) {
+        const nodeNeighbors = links.filter(edge => edge.source === nodeid);
+
+        visited[nodeid] = true;
+        recStack[nodeid] = true;
+
+        for (const neighbor of nodeNeighbors) {
+            if (!visited[neighbor.target] && checkCyclicUtil(links, neighbor.target, visited, recStack)) {
+                return true;
+            } else if (recStack[neighbor.target]) {
+                return true;
+            }
+        }
+    }
+    recStack[nodeid] = false;
+    return false;
+};
+
+/**
+ * Checks if the input nodes and links reflects a cyclic graph.
+ * Note: Makes use of the corresponding checkCyclicUtil function, to perform a recursive dfs search.
+ * @param {Array.<Object>} nodes - list of nodes in a graph.
+ * @param {Array.<Object>} links - list of links connecting the nodes in a graph.
+ * @returns {boolean} boolean reflecting wether the input nodes and links, results in a cyclic graph.
+ * @memberof utils
+ */
+const checkCyclic = (nodes, links) => {
```

Comment:
`utils` is a very core part of the library, thus I ask you to add unit tests for this function.